pkg/test/kube/dump.go

```diff
@@ -152,13 +161,27 @@ func DumpPods(ctx resource.Context, workDir, namespace string, selectors []strin
 const coredumpDir = "/var/lib/istio"
 
 func DumpCoreDumps(ctx resource.Context, c cluster.Cluster, workDir string, namespace string, pods ...corev1.Pod) {
+	if coreDumpedPods.Load() >= maxCoreDumpedPods {
+		return
+	}
 	pods = podsOrFetch(c, pods, namespace)
 	for _, pod := range pods {
+		if coreDumpedPods.Load() >= maxCoreDumpedPods {
+			return
+		}
+		wroteDumpsForPod := false
 		containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
 		for _, container := range containers {
 			if container.Name != "istio-proxy" {
 				continue
 			}
+			restarts := containerRestarts(pod, "istio-proxy")
```

Comment:
Will the core dumps even exist? I'm not sure this check is necessary.

Reply:
There will only be a dump if it crashed I think, so this will reduce exec calls which I think is good.